lib/puppet.rb

```diff
@@ -36,6 +38,27 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    # TODO: (PUP-6958) Handle the rubygems case
+    local_locale_path = File.absolute_path('../locales', File.dirname(__FILE__))
```

Comment:
I'm inclined to suggest we should replace `File.absolute_path` with `Puppet::FileSystem.expand_path` here but I'm curious - was there a reason `File.absolute_path` was selected?